site/src/components/Resources/AgentRow.tsx

```diff
@@ -79,6 +79,11 @@ export const AgentRow: FC<AgentRowProps> = ({
     showApps &&
     ((agent.status === "connected" && hasAppsToDisplay) ||
       agent.status === "connecting");
+  const showVSCode =
+    (agent.display_apps.includes("vscode") ||
+      agent.display_apps.includes("vscode_insiders")) &&
+    !hideVSCodeDesktopButton;
```

Comment:
The display code in this file is fairly complicated - there is a lot of conditional logic and you must understand the difference between the various VS Code apps and the difference between `agent.apps` and `agent.display_apps`. I think we could really benefit from some tests here, and maybe a comment or two. We have some [tests](https://github.com/coder/coder/blob/main/site/src/components/Resources/AgentRowPreview.test.tsx) for very similar logic in the `AgentRowPreview` component that might be worth looking at. Open to storybook tests, too...I'm just not sure how one could describe the business logic adequately through those, but interactions might help out!

Reply:
This is true. Thx for sharing a good test example 🙏

Reply:
I'm not sure the best way to print the results of a "table style" test.. What do you think about this?
![Uploading Screenshot 2024-01-18 at 15.21.33.png…]()